File: 数组/array.php
<?php

$str = "0123456789";
$arr = str_split($str);
print_r($arr);
$first = 0;
$last = count($arr) - 1;
while (true) {
    if ($first < $last) {
        $first_tmp = $arr[$first];
        $arr[$first] = $arr[$last];
        $arr[$last] = $first_tmp;
        $first++;
        $last--;
    } else {
        break;
    }
}
print_r($arr);

File: 数组/array.go
package main

import (
	"fmt"
	"strings"
)

// 字符串翻转(首位交换)
func main() {
	str := "012345678"
	arr := strings.Split(str, "")
	fmt.Println(arr)
	first := 0
	last := len(arr) - 1
	for {
		if first < last {
			arr[first], arr[last] = arr[last], arr[first]
			first++
			last--
		} else {
			break
		}
	}
	str = strings.Join(arr, "")
	fmt.Println(str)
}
